dotnet/src/Experimental/Agents/AgentMessage.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Experimental.Agents;
+
+/// <summary>
+/// Represents agent message used by agents inherited from <see cref="KernelAgent" /> class.
+/// </summary>
+public class AgentMessage
+{
+    /// <summary>
+    /// The inner message representation. Use this to bypass the current abstraction.
+    /// </summary>
+    public object? InnerMessage { get; set; }
+
+    /// <summary>
+    /// Role of the author of the message.
+    /// </summary>
+    public AuthorRole Role { get; set; }
+
+    /// <summary>
+    /// The <see cref="KernelAgent"/> the message was created by.
+    /// </summary>
+    public KernelAgent? Agent { get; set; }
```

Comment:
Will it be practical to marshal an actual agent reference to each message?  (which one?)

Reply:
The scenario I thought about is related to a pre-hook for a chat completion agent that would receive, let's say, an AssistantFileIdContent content type it is not familiar with from an assistant agent. To convert the content type to a BinaryContent, which the chat completion agent can understand, there should be a way to access the message owner (assistant agent) to download the file using the ID. Therefore, having the agent in the message will allow the chat completion agent's pre-hook to download the content by ID, wrap it in BinaryContent, and send it to the chat completion agent who knows how to work with BinaryContent.